src/Utils.ts

```diff
@@ -175,27 +182,38 @@ export const kEmptyObject = Object.freeze(Object.create(null));
 //   return slowCases(enc);
 // }
 
-export function toArrayBuffer(buf: Buffer | SBuffer): ArrayBuffer {
-  if (Buffer.isBuffer(buf) && buf?.buffer?.slice) {
-    return buf.buffer.slice(buf.byteOffset, buf.byteOffset + buf.byteLength);
+export function toArrayBuffer(
+  buf: CraftzdogBuffer | FerossBuffer | SafeBuffer | ArrayBufferView,
+): ArrayBuffer {
+  if (
+    CraftzdogBuffer.isBuffer(buf) ||
+    FerossBuffer.isBuffer(buf) ||
+    ArrayBuffer.isView(buf)
+  ) {
+    if (buf?.buffer?.slice) {
+      return buf.buffer.slice(buf.byteOffset, buf.byteOffset + buf.byteLength);
+    } else {
+      return buf.buffer;
```

Comment:
When does this happen? If it does happen (if a Buffer or ArrayBufferView does not implement slice), I'd throw an error instead as this is unexpected.

Reply:
agreed.  Throwing an error.